content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1007 @@
+
+Hopefully by now you're convinced that this is the right move. If not ... I'm sorry ... you still need to implement Component manually in Bevy 0.6. You can either derive Component:
+
+```rust
+// defaults to "Table" storage
+#[derive(Component)]
+struct SomeComponent;
+
+// overrides the default storage
+#[derive(Component)]
+#[component(storage = "SparseSet")]
+struct SomeComponent;
+```
+
+Or you can manually implement it:
+
+```rust
+struct SomeComponent;
+
+impl Component for SomeComponent {
+    type Storage = TableStorage;
+}
+
+impl Component for Movable {
+    type Storage = SparseSetStorage;
+}
+```
+
+### iter() for mutable Queries
+
+<div class="release-feature-authors">authors: @Guvante</div>
+
+Mutable queries can now be immutably iterated (which returns immutable references):
+
+```rust
+fn system(mut players: Query<&mut Player>) {
+    for player in players.iter() {
+        // player is an immutable reference
+    }
+
+    for mut player in players.iter_mut() {
+        // player is a mutable reference
+    }
+}
+```
+
+Compare that to the complicated QuerySet that this would have needed in previous Bevy versions to avoid conflicting immutable and mutable Queries:
+
+```rust
+// Gross!
+fn system(mut players: QuerySet<(QueryState<&Player>, QueryState<&mut Player>)>) {
+    for player in players.q0().iter() {
+        // player is an immutable reference
+    }
+
+    for mut player in players.q1().iter_mut() {
+        // player is a mutable reference
+    }
+}
+```
+
+### SystemState
+
+Have you ever wanted to use "system params" directly with a Bevy World? With {{rust_type(type="struct" crate="bevy_ecs" version="0.6.0" name="SystemState")}}, now you can!
+
+```rust
+let mut system_state: SystemState<(Res<A>, Query<&B>)> = SystemState::new(&mut world);
+let (a, query) = system_state.get(&world);
+```
+
+{{rust_type(type="struct" crate="bevy_ecs" version="0.6.0" name="SystemState")}} does all of the same caching that a normal Bevy system does, so reusing the same SystemState results in uber-fast World access.
+
+### Sub Apps
+
+<div class="release-feature-authors">authors: @cart, @zicklag, @bjorn3</div>
+
+The new Bevy renderer requires strict separation between the "main app" and the "render app". To enable this, we added the concept of "sub apps":
+
+```rust
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, AppLabel)]
+pub struct RenderApp;
+
+let mut render_app = App::empty();
+app.add_sub_app(RenderApp, render_app, move |app_world, render_app| {
+    // execute app logic here
+});
+
+// later
+app.sub_app_mut(RenderApp)
+    .add_system(some_system);
+    .add_system(some_other_system);
+```
+
+We plan on exposing more control over scheduling and running sub apps in the future.
+
+### Query::iter_combinations
+
+<div class="release-feature-authors">authors: @Frizi</div>
+
+You can now iterate all permutations of N entities for a given query:
+
+```rust
+fn system(query: Query<&Player>) {
```

Comment:
This should be a mutable reference, as that's where the true power of the feature lies.

Reply:
That complicates things slightly as then we need to go in to why you cant use normal iterators with the mutable variant / why the apis aren't consistent.  I think I'd rather let documentation + examples cover that use case, just to save space (and user brain power).